src/experimental/standalone_pem/standalone_pem_manager.cc

```diff
@@ -164,32 +164,53 @@ Status StandalonePEMManager::InitSchemas() {
   int64_t stirling_error_table_size = FLAGS_table_store_stirling_error_limit_bytes / 2;
   int64_t probe_status_table_size = FLAGS_table_store_stirling_error_limit_bytes / 2;
   int64_t proc_exit_events_table_size = FLAGS_table_store_proc_exit_events_limit_bytes;
-  int64_t other_table_size = (memory_limit - http_table_size - stirling_error_table_size -
-                              probe_status_table_size - proc_exit_events_table_size) /
-                             (num_tables - 4);
 
+  // Determine which of the four default tables are present
+  bool has_http_events = false, has_stirling_error = false, has_probe_status = false,
+       has_proc_exit_events = false;
+  for (const auto& relation_info : relation_info_vec) {
+    if (relation_info.name == "http_events")
+      has_http_events = true;
+    else if (relation_info.name == "stirling_error")
+      has_stirling_error = true;
+    else if (relation_info.name == "probe_status")
+      has_probe_status = true;
+    else if (relation_info.name == "proc_exit_events")
+      has_proc_exit_events = true;
+  }
+
+  // Calculate memory used by specific tables
+  int64_t used_memory = 0;
+  if (has_http_events) used_memory += http_table_size;
+  if (has_stirling_error) used_memory += stirling_error_table_size;
+  if (has_probe_status) used_memory += probe_status_table_size;
+  if (has_proc_exit_events) used_memory += proc_exit_events_table_size;
+
+  int64_t remaining_memory = memory_limit - used_memory;
+  int64_t other_table_count =
+      num_tables - (has_http_events + has_stirling_error + has_probe_status + has_proc_exit_events);
+  int64_t other_table_size = (other_table_count > 0) ? remaining_memory / other_table_count : 0;
+
+  // Create tables with allocated sizes
   for (const auto& relation_info : relation_info_vec) {
     std::shared_ptr<table_store::Table> table_ptr;
+    int64_t table_size = other_table_size;  // Default size
+
     if (relation_info.name == "http_events") {
       // Special case to set the max size of the http_events table differently from the other
       // tables. For now, the min cold batch size is set to 256kB to be consistent with previous
       // behaviour.
-      table_ptr = std::make_shared<table_store::Table>(relation_info.name, relation_info.relation,
-                                                       http_table_size, 256 * 1024);
```

Comment:
The http compacted batch size of `256 * 1024` is no longer being applied. Was that intentional?

I think we should maintain the same behavior unless there is a strong reason to change it.

Reply:
Thanks for catching that! I agree, we should keep the http batch size. Fixed in latest commit.